std/assembly/util/number.ts

```diff
@@ -139,7 +164,22 @@ export function decimalCount64High(value: u64): u32 {
   }
 }
 
-function utoa32_lut(buffer: usize, num: u32, offset: usize): void {
+// TODO: improve this
+function ulogBase(n: u64, base: i32): u32 {
+  // // this approsh required overflow checking
+  // var b = u64(base), t = b, e: u32 = 1;
+  // while (t <= n) {
+  //   t *= b; e++;
+  // }
+  // return e;
+  var b = u64(base), e: u32 = 1;
+  while (n > 1) {
+    n /= b; e++;
+  }
+  return e;
+}
```

Comment:
That's hardest and inefficient part for general case.
It's quite hard implement integer logarithm with any base efficiently. It's problem pretty similar to discrete logarithm except we don't care about prime modules.

@jedisct1 may be you suggest something?

I really don't like do twice allocations extra loop for string reversing if we don't pre-calculate number of decimals via ulogBase.